File: java/business/src/main/java/business/services/TeamService.java
package business.services;

import repository.interfaces.TeamRepository;
import models.Team;

public class TeamService {
    private TeamRepository teamRepository;

    public TeamService(TeamRepository teamRepository) {
        this.teamRepository = teamRepository;
    }

    public Team addTeam(String teamName) {
        Team team = new Team(teamName);
        return teamRepository.save(team);
    }

}

File: csharp/Business/Services/TeamService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


using Repository.Interfaces;
using Domain.Models;

namespace Business.Services
{
    public class TeamService
    {
        private ITeamRepository teamRepository;

        public TeamService(ITeamRepository teamRepository)
        {
            this.teamRepository = teamRepository;
        }

        public Team AddTeam(String teamName)
        {
            Team team = new Team(teamName);
            return teamRepository.Add(team);
        }

        public ICollection<Team> GetAllTeams()
        {
            return teamRepository.FindAll();
        }

        public Team FindTeamByName(String teamName)
        {
            return teamRepository.FindTeamByName(teamName); 
        }
    }
}
